File: src/main/java/la/jitco/jana/Delay.java
package la.jitco.jana;

public class Delay extends Module {
    
    Delay(int id)
    {
        super(id);
        type = ModuleType.DELAY;
        name = "Delay " + id;
        paramArray = new Parameter[] {
                         new PositiveFloatingPointParameter("initTime"),
                         new NonnegativeFloatingPointParameter("initFeedback"),
                         new NonnegativeFloatingPointParameter("initLevel")};
        modInArray = new ModIn[] {new ModIn(this, "input"),
                                  new ModIn(this, "time"),
                                  new ModIn(this, "feedback"),
                                  new ModIn(this, "level")};
        modOutArray = new ModOut[] {new ModOut(this, "mainOut")};
        createMaps();
        weightTaskMultiplier = 5; // TODO: find actual value
        setWeight();
    }
    
}

File: JobProcessing/Delay.cpp
#include "Delay.h"

#include <string>
#include "ModIn.h"
#include "ModOut.h"

Parameter& Delay::getParameter(std::string parameterName)
{
    if (parameterName == "initTime")
        return initTime;
    else if (parameterName == "initFeedback")
        return initFeedback;
    else if (parameterName == "initLevel")
        return initLevel;
    else
        return initTime;
}


ModIn& Delay::getModIn(std::string modInName)
{
    if (modInName == "input")
        return input;
    else if (modInName == "time")
        return time;
    else if (modInName == "feedback")
        return feedback;
    else if (modInName == "level")
        return level;
    else
        return input;
}

ModOut& Delay::getModOut(std::string modOutName)
{
    if (modOutName == "mainOut")
        return mainOut;
    else
        return mainOut;
}
